node/src/accountant/mod.rs

```diff
@@ -1882,32 +1840,43 @@ mod tests {
+        let pending_payable_begin_scan_params_arc = Arc::new(Mutex::new(vec![]));
+        let receivable_begin_scan_params_arc = Arc::new(Mutex::new(vec![]));
+        let payable_scanner = ScannerMock::new().begin_scan_params(&payable_begin_scan_params_arc);
+        let pending_payable_scanner =
+            ScannerMock::new().begin_scan_params(&pending_payable_begin_scan_params_arc);
+        let receivable_scanner =
+            ScannerMock::new().begin_scan_params(&receivable_begin_scan_params_arc);
         let mut config = bc_from_earning_wallet(make_wallet("hi"));
         config.scan_intervals_opt = Some(ScanIntervals {
-            payable_scan_interval: Duration::from_millis(1),
-            receivable_scan_interval: Duration::from_millis(1),
-            pending_payable_scan_interval: Duration::from_secs(100),
+            payable_scan_interval: Duration::from_millis(100),
+            receivable_scan_interval: Duration::from_millis(100),
+            pending_payable_scan_interval: Duration::from_millis(100),
         });
-        config.suppress_initial_scans_opt = Some(true);
-        let payable_dao = PayableDaoMock::new(); // No payables: demanding one would cause a panic
-        let receivable_dao = ReceivableDaoMock::new(); // No delinquencies: demanding one would cause a panic
+        config.suppress_initial_scans = true;
         let peer_actors = peer_actors_builder().build();
-        let subject = AccountantBuilder::default()
+        let mut subject = AccountantBuilder::default()
             .bootstrapper_config(config)
-            .payable_dao(payable_dao) // For Accountant
-            .payable_dao(PayableDaoMock::new()) // For Payable Scanner
-            .payable_dao(PayableDaoMock::new()) // For PendingPayable Scanner
-            .receivable_dao(receivable_dao) // For Accountant
-            .receivable_dao(ReceivableDaoMock::new()) // For Scanner
             .build();
+        subject.scanners.payable = Box::new(payable_scanner);
+        subject.scanners.pending_payable = Box::new(pending_payable_scanner);
+        subject.scanners.receivable = Box::new(receivable_scanner);
         let subject_addr = subject.start();
         let subject_subs = Accountant::make_subs_from(&subject_addr);
         send_bind_message!(subject_subs, peer_actors);
 
         send_start_message!(subject_subs);
 
         System::current().stop();
-        assert_eq!(system.run(), 0);
+        system.run();
+        let payable_begin_scan_params = payable_begin_scan_params_arc.lock().unwrap();
+        let pending_payable_begin_scan_params =
+            pending_payable_begin_scan_params_arc.lock().unwrap();
+        let receivable_begin_scan_params = receivable_begin_scan_params_arc.lock().unwrap();
+        assert_eq!(payable_begin_scan_params.is_empty(), true);
+        assert_eq!(pending_payable_begin_scan_params.is_empty(), true);
+        assert_eq!(receivable_begin_scan_params.is_empty(), true);
```

Comment:
Again I feel like a 🎾, I thought you wanted that, that's why I added it. Are you sure, it would hurt the codebase if I'll remove it now?

Reply:
Hmm...

I felt guilty for what you are referring me to. So I went to check what I actually instructed in the review 0 (well, this one is called 1 but is 2 correctly). I didn't find there anything that should have led you to rearrangement which was done later on. I don't know where the idea comes from.

However, I think that the scanners are probably (or should be) asserted with the params somewhere else where it is more needed. I suppose we could see it where the cycles of scanning are repeating and we test that we went through at least two cycles. 

This kind of test, which we discuss here, doesn't require such heavy tooling though. It really will be as much provable as it is without the param assertions. 

Again, the trick lies in on-purpose undelivered results for the used mocks. If you do that, plus if you make it obvious (sometimes we write a comment, which you can see, by the way, in the older version, written by Dan if remember). So you can extend the comment or adjust a little but otherwise the principle is just the same. 
We run it, no new scans are requested, the Actix system doesn't need to continue running and stops. However it stops clear, without a panic (and so the assertion `assert_eq!(system.run(), 0)` should be in place). 

I'm sorry if you feel injustice in what I've suggested but I do think my reasons are grounded. :pray:

Reply:
This was the comment - https://github.com/MASQ-Project/Node/pull/176#discussion_r1029482734

Reply:
Oh yeah. 

I think I will stop abusing you in the concern of this card :slightly_smiling_face: 

However, I still think there is no big difference between my [and it's not just mine; it was actually Dan who invented that and I just found it cool] and your approach. Yours is more explicit but also wordy.
 
At least I'd like you to accept what I suggested into your dev toolset, because I'm positive we will continue using it in some tests, me and Dan, and I don't see why you should turn this method into the unacceptable. It's simply legitimate. Dan sometimes says "_Hmm, no, we can do even better_...which means he's planning to remove the assertions and leave the test being based on the implicit thought construct. 

Just to make sure you understood me, this is how it works: 
If you have a test where you use mocks and the mocks are constructed the standard way (the should be), it means that if you want to call any possible method from the mock's trade, you will have put up results for it as long as you want to avoid panicking on calling the method and internally calling .remove(0) on an empty queue of results. 

This principle is the core of the idea that at some tests you don't need to use `assertxx` macros. 
(Then it's only politeness to tell about it in a comment to just navigate the reader a bit but I think he could understand even alone).

If I'm to generalize on this, this method is useful in all those places where we try to prove that _something didn't run_. That is also this test. And you could see that Dan also initially wrote the test the way without explicit assertions.

I won't be angry at you, but I'm saying I wouldn't hesitate to remove it. I thought a lot about potential benefits of your assertions and found none except it's literal. I just want to encourage you to rather stop using it in the future. 

---

That was first part. However, we need to deal with the comment as I don't like there are your assertions and the comment together. I think you should delete it.

Also change the log assertion to assert on the name of the log. Make the name of the logger sound the same as the name of the test please.